nntrainer/app_context.cpp

```diff
@@ -107,32 +207,39 @@ static void add_default_object(AppContext &ac) {
                      LayerType::LAYER_UNKNOWN);
 }
 
-AppContext &AppContext::Global() {
-  static AppContext instance;
-  std::call_once(global_app_context_init_flag, add_default_object,
-                 std::ref(instance));
-  return instance;
-}
+static void add_extension_object(AppContext &ac) {
+  auto dir_list = getPluginPaths();
 
-static const std::string func_tag = "[AppContext] ";
-
-namespace {
-const std::string getFullPath(const std::string &path,
-                              const std::string &base) {
-  /// if path is absolute, return path
-  if (path[0] == '/') {
-    return path;
+  for (auto &path : dir_list) {
+    try {
+      ac.registerLayerFromDirectory(path);
```

Comment:
from the point of view of this function, shouldn't this just call a hypothetical `ac.registerFromDirectory(path)`. `registerFromDirectory` can figure out if the file inside is a layer or other type based on prefix?

Reply:
What you've said sounds about right.

We could have

```
void registerFromDirectory(path) {
ac.registerLayerFromDirectory(path);
ac.registerOptimizerFromDirectory(path);
ac.registerModelsFromDirectory(path);
}
```

or if we fix the policy as having only one folder and put everything inside. below could work too.
```
void registerFromDirectory(path, const std::string prefix = "_layer.so") {
}
```

It seems more straight forward to have `ac.registerLayerFromDirectory` for now, later extend in a way you recommended when we start extending the feature to other modules after fixing some policies related to this.